core/common/src/main/java/alluxio/heartbeat/HeartbeatThread.java

```diff
@@ -123,48 +146,29 @@ public void run() {
       while (!Thread.interrupted()) {
         // TODO(peis): Fix this. The current implementation consumes one thread even when ticking.
         mStatus = Status.WAITING;
-        mTimer.tick();
+        long limitTime = mTimer.tick();
         mStatus = Status.RUNNING;
-        mExecutor.heartbeat();
+        LOG.debug("{} #{} will run limited in {}s", mThreadName, counter++, limitTime / 1000);
+        mExecutor.heartbeat(limitTime);
```

Comment:
Who stops this heartbeat if the limitTime has been exceeded?

Reply:
stopped by Heartbeat implementation